spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Recurrent.scala

```diff
@@ -260,7 +293,13 @@ class Recurrent[T : ClassTag]()
       currentGradOutput(inputDim) = gradOutput.select(timeDim, i)
       _input(hidDim) = if (i > 1) cells(i - 2).output.toTable(hidDim)
         else hidden
-      _input(inputDim) = outputCell.select(timeDim, i)
+
+      _input(inputDim) = if (feedbackOutput && preTopology == null) {
+        newInput.select(timeDim, i)
+      } else {
+        outputCell.select(timeDim, i)
+      }
```

Comment:
Such if-else condition will make code hard to rebase and review.
Split preTopology and feedbackOutput into two different branch?